crates/router/src/core/payments.rs

```diff
@@ -210,6 +222,17 @@ where
             .await?;
     }
 
+    #[cfg(feature = "locking")]
+    let _ = locking::release_lock(state, 3, acquired_lock.to_owned())
```

Comment:
Will it always reach here?, I mean, even in case of any errors inside the flow?. We should never fail to release the lock

Reply:
Let's add metrics in case of failure. And monitor.

Reply:
Even in case of error the payments flow completes the post_update_trackers so that the payment status should be in a conclusive state. But if we are to use in any other flows we have to make sure the flow completes . If not handle release in the parent function based on result being `Ok` or `Err`. 

> Let's add metrics in case of failure. And monitor.

Will be adding metrics in the next PR. added it in TODO.